File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum =0;
        int n = mat.length;
        for(int i=0;i<n;i++){
            sum += mat[i][i] + mat[i][n-1-i];
        }
        if(n%2==0){
            return sum;
        }
        return sum-=mat[n/2][n/2];
       
    }
}

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function(mat) {
    let sum = 0;
    let n = mat.length;
    for(let i=0;i<n;i++){
        sum+=mat[i][i]+mat[i][n-1-i];
    }
    if(n%2 != 0){
        let op = Math.floor(n/2);
        return sum-=mat[op][op];
    }
    return sum;
};